langkit/pii.py

```diff
@@ -0,0 +1,115 @@
+from copy import deepcopy
+from presidio_analyzer import AnalyzerEngine, RecognizerResult
+from whylogs.experimental.core.udf_schema import (
+    register_multioutput_udf,
+)
+import pandas as pd
+from typing import Dict, List, Optional, Tuple
+from langkit import LangKitConfig, lang_config, prompt_column, response_column
+from langkit.pattern_loader import PresidioEntityLoader
+from langkit.utils import _unregister_metric_udf
+import json
+
+_registered: List[str] = []
+
+entity_loader = PresidioEntityLoader()
+
+
+# entities = ["PHONE_NUMBER", "US_PASSPORT"]
+analyzer = AnalyzerEngine()
+
+
+def format_presidio_result(result: RecognizerResult) -> dict:
+    return {
+        "type": f"{result.entity_type}",
+        "start": f"{result.start}",
+        "end": f"{result.end}",
+        "score": f"{result.score}",
+    }
+
+
+def analyze_pii(text: str) -> Tuple[str, int]:
+    global analyzer
+    global entity_loader
+
+    entities = entity_loader.get_entities()
+    results = analyzer.analyze(
+        text=text,
+        entities=entities,
+        language="en",
+    )
+    dict_results = [format_presidio_result(entity) for entity in results]
+    return (json.dumps(dict_results), len(dict_results))
+
+
+def _wrapper(column):
+    def wrappee(text):
+        analyzer_results: List[tuple] = []
+        for input in text[column]:
+            analyzer_results.append(analyze_pii(input))
+        if isinstance(text, pd.DataFrame):
+            return pd.DataFrame(
+                {
+                    "result": [x[0] for x in analyzer_results],
+                    "entities_count": [x[1] for x in analyzer_results],
```

Comment:
suggest making these constants since you use them more than once.

Reply:
resolved